lib/cylc/task_events_mgr.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2017 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Task events manager."""
+
+import os
+import shlex
+from time import time
+import traceback
+
+from cylc.cfgspec.globalcfg import GLOBAL_CFG
+import cylc.flags
+from cylc.mp_pool import SuiteProcContext
+from cylc.suite_logging import ERR, LOG
+from cylc.task_proxy import TaskProxy
+from cylc.task_state import TASK_STATUS_SUCCEEDED
+
+
+class TaskEventsManager(object):
+    """Task events manager."""
```

Comment:
(the doc strings would seem more helpful if the class names were less helpful 😁)